src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Utilities/ValueTaskExtensions.cs

```diff
@@ -0,0 +1,29 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Threading.Tasks;
+
+namespace Roslyn.Utilities
+{
+    internal static class ValueTaskExtensions
+    {
+        /// <summary>
+        /// Asserts the <see cref="ValueTask"/> passed has already been completed.
+        /// </summary>
+        /// <remarks>
+        /// This is useful for a specific case: sometimes you might be calling an API that is "sometimes" async, and you're
+        /// calling it from a synchronous method where you know it should have completed synchronously. This is an easy
+        /// way to assert that while silencing any compiler complaints.
+        /// </remarks>
+        public static void AssertCompleted(this ValueTask valueTask)
+        {
+            Contract.ThrowIfFalse(valueTask.IsCompleted);
```

Comment:
📝 This needs to be wrapped in a `try`/`catch` to ensure the result is not lost

```suggestion
            try
            {
                Contract.ThrowIfFalse(valueTask.IsCompleted);
            }
            catch
            {
                valueTask.Preserve();
                throw;
            }
```

Reply:
Can you clarify why this is supposed to help?

Reply:
You still need to clean up the task if the precondition fails.

Reply:
Chatting with @sharwell privately we figured just replacing this with a regular Task is probably the right thing to do; I chose ValueTask as "why not?" and if the answer is we're having to get into the weeds of ValueTask then we have a good answer for "why not?". This is somewhat supported by the comments and discussion in https://github.com/microsoft/vs-threading/issues/556.